src/main/java/com/uid2/admin/vertx/service/SiteService.java

```diff
@@ -389,6 +439,23 @@ private static List<String> getNormalizedDomainNames(RoutingContext rc, JsonArra
         return normalizedDomainNames;
     }
 
+    private static List<String> getNormalizedAppNames(RoutingContext rc, JsonArray appNamesJa) {
+        List<String> appNames = appNamesJa.stream().map(String::valueOf).collect(Collectors.toList());
+
+        List<String> normalizedAppNames = new ArrayList<>();
+        for (String appName : appNames) {
```

Comment:
you can probably combine the stream operation with the for loop
```
List<String> normalizedAppNames = appNamesJa.stream()
                .map(String::valueOf)
                .map(String::toLowerCase)
                .collect(Collectors.toList());
```

Reply:
That's is a good idea. We no longer need `toLowerCase` method but I will do it next time.